test/metabase/sync/sync_metadata/fields/sync_metadata_test.clj

```diff
@@ -41,13 +43,33 @@
              :database-type     "Integer"
              :base-type         :type/Integer
              :database-position 0
-             :database-required false}
+             :database-required false
+             :database-is-auto-increment false}
             {:name              "My Field"
              :database-type     "Integer"
              :base-type         :type/Integer
              :id                1
              :database-position 0
-             :database-required true})))))
+             :database-required true
+             :database-is-auto-increment false})))))
+
+(deftest database-is-auto-increment-changed-test
+  (testing "test that if database-required changes we will update it in the DB"
+    (is (= [["Field" 1 {:database_is_auto_increment true}]]
+           (updates-that-will-be-performed
+            {:name              "My Field"
+             :database-type     "Integer"
+             :base-type         :type/Integer
+             :database-position 0
+             :database-required false
+             :database-is-auto-increment true}
+            {:name              "My Field"
+             :database-type     "Integer"
+             :base-type         :type/Integer
+             :id                1
+             :database-position 0
+             :database-required false
+             :database-is-auto-increment false})))))
```

Comment:
Okay, you need to add a test here for the case where `database-is-auto-increment=nil` then